sdcm/nemesis.py

```diff
@@ -2784,12 +2783,54 @@ def disrupt_network_start_stop_interface(self):  # pylint: disable=invalid-name
             self.target_node.start_network_interface()
             self._wait_all_nodes_un()
 
-    def break_streaming_task_and_rebuild(self, task='decommission'):  # pylint: disable=too-many-statements
+    def _call_disrupt_func_after_expression_logged(self,
+                                                   expression: str,
+                                                   func: Callable,
+                                                   func_kwargs: dict = None,
+                                                   sleep: int = 1,
+                                                   delay: int = 10,
+                                                   timeout: int = 600):
         """
-        Stop streaming task in middle and rebuild the data on the node.
+        This method watches the target node logs for an expression
+        with a <sleep> time step. Once the expression is found it
+        will call the callable <func> with <func_kwargs> keyword
+        arguments after a <delay>.
         """
+        start_time = time.time()
+        target_node_logs = self.target_node.follow_system_log(patterns=[expression])
 
+        with DbEventsFilter(db_event=DatabaseLogEvent.RUNTIME_ERROR,
+                            line="This node was decommissioned and will not rejoin",
+                            node=self.target_node), \
+            DbEventsFilter(db_event=DatabaseLogEvent.RUNTIME_ERROR,
+                           line="Fail to send STREAM_MUTATION_DONE",
+                           node=self.target_node), \
+            DbEventsFilter(db_event=DatabaseLogEvent.DATABASE_ERROR,
+                           line="streaming::stream_exception",
+                           node=self.target_node), \
+            DbEventsFilter(db_event=DatabaseLogEvent.RUNTIME_ERROR,
+                           line="got error in row level repair",
+                           node=self.target_node):
+            while time.time() - start_time < timeout:
+                if target_node_logs:
```

Comment:
To be honest I think we don't - this should end up dealing with a Tuple, but I added the cast to `list` either way, since it might save us some code digging down the line